### DISCUSSION TITLE
Not sure if extensions works

### DISCUSSION THREAD

**Discussion by ghost** [swiftlang/vscode-swift#330] [Q&A] [answered]
Hi, I installed the extension in VScode (on Linux, version 1.68.1).

The plugin does not seem to work at all, even after me editing some of the settings of the plugin to make sure it found the toolchain correctly

Nothing happens whenever I open either my Package.swift file or even my main.swift file,
I do not have any "Package dependencies" view getting added to the file explorer
I do not have any of `Resolve Package Dependencies`, `Update Package Dependencies`, `Reset Package Dependencies` in the command palette
And I also do not have any completion whatsoever when trying to type code (besides the basic one from vscode itself)

**Comment by 0xTim** [comment, marked as answer]
@InfRandomness which version of the extension do you have installed? And does `swift --version` work from the command line?

If you go to the Output window in VSCode and select the Swift option from the dropdown menu - what do you see? Just to be clear are you opening individual files or the project directory?

**Reply by adam-fowler** [reply]
Also can you give me a little more information about your system

 - OS: [e.g. Ubuntu 20.04]
 - Swift version (output of `swift --version`)

**Reply by ghost** [reply]
> @InfRandomness which version of the extension do you have installed? And does `swift --version` work from the command line?
> 
> If you go to the Output window in VSCode and select the Swift option from the dropdown menu - what do you see? Just to be clear are you opening individual files or the project directory?

Sorry for the lack of infos : 
my swift version is `Swift version 5.6.1 (swift-5.6.1-RELEASE)`
my OS is Fedora 36

> what do you see?

Nothing relevant coming from the extension

> Just to be clear are you opening individual files or the project directory?

I'm opening the directory

**Reply by adam-fowler** [reply]
I haven't been able to test Fedora. What happens when you type `/bin/sh -c "type swift"` in a terminal?

**Reply by ghost** [reply]
> /bin/sh -c "type swift"

```
[infrandomness@fedora src]$ /bin/sh -c "type swift"
swift est /usr/bin/swift
```

**Reply by adam-fowler** [reply]
Ah I know what it is, I've been caught out by localisation of linux output. I assumed it would return `swift is ...`. Sorry I'll add a bug to fix this.

**Reply by ghost** [reply]
> I haven't been able to test Fedora. What happens when you type `/bin/sh -c "type swift"` in a terminal?

Wait wait,
it seems to work now
I just restarted my computer

Also this output is normal, my locales are in French

**Reply by adam-fowler** [reply]
You should be able to override the search for swift by setting `swift.path` to `/usr/bin/`

**Reply by adam-fowler** [reply]
Could you verify for me what the output of `LC_MESSAGES=C /bin/sh -c "type swift"` is

**Reply by ghost** [reply]
> LC_MESSAGES=C /bin/sh -c "type swift"

```
[infrandomness@fedora hooks]$ LC_MESSAGES=C /bin/sh -c "type swift"
swift is /usr/bin/swift
```

**Comment by aelam** [comment]
#387 Please check this one, probably it can fix this issue
